anta/tests/path_selection.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""Test functions related to various router path-selection settings."""
+
+# Mypy does not understand AntaTest.Input typing
+# mypy: disable-error-code=attr-defined
+from __future__ import annotations
+
+from ipaddress import IPv4Address
+from typing import ClassVar
+
+from pydantic import BaseModel
+
+from anta.models import AntaCommand, AntaTemplate, AntaTest
+from anta.tools import get_value
+
+
+class VerifyRouterPathsHealth(AntaTest):
+    """
+    Verifies the route and telemetry state of all paths under router path-selection.
+
+    The expected states are 'IPsec established', 'Resolved' for route and 'active' for telemetry.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if all paths under router path-selection have their route state as either 'IPsec established' or 'Resolved'
+               and their telemetry state as 'active'.
+    * Failure: The test will fail if router path-selection is not configured, any path's route state is not 'IPsec established' or 'Resolved',
+               or the telemetry state is 'inactive'.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.path_selection:
+      - VerifyRouterPathsHealth:
+    ```
+    """
+
+    name = "VerifyRouterPathsHealth"
+    description = "Verifies the route and telemetry state of all paths under router path-selection."
+    categories: ClassVar[list[str]] = ["path-selection"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show path-selection paths")]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyRouterPathsHealth."""
+        self.result.is_success()
+
+        command_output = self.instance_commands[0].json_output["dpsPeers"]
+
+        # If no paths are configured for router path-selection, the test fails
+        if not command_output:
+            self.result.is_failure("No paths are configured for router path-selection.")
+            return
+
+        # Check the state of each path
+        for peer, peer_data in command_output.items():
+            for group, group_data in peer_data["dpsGroups"].items():
+                for path_data in group_data["dpsPaths"].values():
+                    route_state = path_data["state"]
+                    session = path_data["dpsSessions"]["0"]["active"]
+
+                    # If the route state of any path is not 'ipsecEstablished' or 'routeResolved', the test fails
+                    if route_state not in ["ipsecEstablished", "routeResolved"]:
+                        self.result.is_failure(f"Route state for peer {peer} in group {group} is `{route_state}`.")
```

Comment:
so here we could even imagine checking that if IPsec is enabled for the path-group for dynamic peers, we expect ipsecEstablished otherwise routeResolved. It would mean also retrieving the `show running-config path-seleciton